### PR TITLE
feat: Add support for EC2 reserved instances

### PR DESCRIPTION
**Pull request by hermanschaaf**
This adds support for fetching EC2 reserved instances. 

It is meant to address https://github.com/cloudquery/cloudquery/issues/1815, but please note that since this change is being applied to (the as-of-yet unreleased) CloudQuery v2, it will require upgrading to the new version once it is released.

I couldn't see a way to paginate through these results, so I hope that the API returns the full resultset for each region in the first call.

### CODE REVIEW

File: plugins/source/aws/resources/services/ec2/reserved_instances_fetch.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	svc := c.Services().EC2
+
+	// this API does not seem to support any form of pagination
+	output, err := svc.DescribeReservedInstances(ctx, &config, func(options *ec2.Options) {
+		options.Region = c.Region
+	})
+	if err != nil {
+		return err
+	}
+	for _, instance := range output.ReservedInstances {
```
Comment (disq) [inline plugins/source/aws/resources/services/ec2/reserved_instances_fetch.go:26]: res <- output.ReservedInstances

File: plugins/source/aws/resources/services/ec2/reserved_instances_fetch.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+func fetchEc2ReservedInstances(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	var config ec2.DescribeReservedInstancesInput
+	c := meta.(*client.Client)
+	svc := c.Services().EC2
+
+	// this API does not seem to support any form of pagination
+	output, err := svc.DescribeReservedInstances(ctx, &config, func(options *ec2.Options) {
+		options.Region = c.Region
```
Comment (disq) [inline plugins/source/aws/resources/services/ec2/reserved_instances_fetch.go:21]: We shouldn't need to set the region here.

**Review by disq — CHANGES_REQUESTED** [review]

**Review by bbernays — CHANGES_REQUESTED** [review]
I think you need to add the function to `plugins/source/aws/resources/plugin/tables.go` and then add the docs

File: plugins/source/aws/resources/services/ec2/reserved_instances_mock_test.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package ec2
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
```
Comment (disq) [inline plugins/source/aws/resources/services/ec2/reserved_instances_mock_test.go:7]: nit:
```suggestion
	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
```

**Review by disq — APPROVED** [review]

File: plugins/source/aws/resources/services/ec2/reserved_instances_mock_test.go (hunk 3 above)
Comment (hermanschaaf) [reply plugins/source/aws/resources/services/ec2/reserved_instances_mock_test.go]: Fixed here and in a number of other places

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**